_posts/2018-12-06.Rust-1.31.md

```diff
@@ -0,0 +1,338 @@
+have deterministic output. This is important for soundness reasons.
+Currently, `const fn`s can do a minimal subset of operations. Here's
+some examples of what you can do:
+
+* Arithmetic and comparison operators on integers
+* All boolean operators except for `&&` and `||`
+* Constructing arrays, structs, enums, and tuples
+* Calls to other `const fn`s
+* Index expressions on arrays and slices
+* Field accesses on structs and tuples
+* Reading from constants (but not statics, not even taking a reference to a static)
+* `&` and `*` of references
+* Casts, except for raw pointer to integer casts
+
+We'll be slowly growing the abilities of `const fn`, but we've decided that
+this is enough useful stuff to start shipping the feature itself.
+
+For full details, please see [the
+reference](https://doc.rust-lang.org/reference/items/functions.html#const-functions).
+
+### Tool lints
+
+In [Rust 1.30](https://blog.rust-lang.org/2018/10/25/Rust-1.30.0.html), we
+stabilized "tool attributes", like `#[rustfmt::skip]`. In Rust 1.31, we're
+stabilizing something similar: "tool lints," like
+`#[allow(clippy::something)]` These give a namespace to lints, so that it's
+more clear which tool they're coming from.
+
+If you previously used Clippy's lints, you can migrate like this:
+
+```rust
+// old
+#![allow(bool_comparison)]
+
+// new
+#![allow(clippy::bool_comparison)]
+```
+
+You'll get warnings that can help you update to the new style.
+
+### Library stabilizations
+
+A bunch of `From` implementations have been added:
+
+* `u8` now implements `From<NonZeroU8>`, and likewise for the other numeric types and their `NonZero` equivalents
+* `Option<&T>` implements `From<&Option<T>>`, and likewise for `&mut`
+
+Additionally, these functions have been stabilized:
+
+* [`slice::align_to`](https://doc.rust-lang.org/std/primitive.slice.html#method.align_to) and its mutable counterpart
+* [`slice::chunks_exact`](https://doc.rust-lang.org/std/primitive.slice.html#method.chunks_exact),
+  as well as its mutable and `r` counterparts (like
+  [`slice::rchunks_exact_mut`](https://doc.rust-lang.org/std/primitive.slice.html#method.rchunks_mut)) in all combinations
+
+See the [detailed release notes][notes] for more.
+
+### Cargo features
+
+Cargo will now download packages in parallel using HTTP/2. Additionally,
+now that `extern crate` is gone, you can't `extern crate foo as bar;` anymore
+to rename a crate. As such, you can do so in your `Cargo.toml`, like this:
+
+```toml
+[dependencies]
+baz = { version = "0.1", package = "foo" }
+```
+
+or, the equivalent
+
+```toml
+[dependencies.baz]
+version = "0.1"
+package = "foo"
+```
+
+Now, the `baz` package will be able to be used via `foo` in your code.
+
+See the [detailed release notes][notes] for more.
+
+## Contributors to 1.31.0
+
+Many people came together to create Rust 1.31. We couldn't have done it
+without all of you. [Thanks!](https://thanks.rust-lang.org/rust/1.31.0)
```

Comment:
Link doesn't seem to work?

Reply:
Yes, that's expected (we run thanks during release).